### PR TITLE
make modform rings for triangle groups use Algebras

### PR DESCRIPTION
**Pull request by fchapoton**
This is using the category framework in a few more rings, namely modular forms for triangle groups.

### :memo: Checklist

- [ ] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/modular/modform_hecketriangle/abstract_ring.py [hunk 1]
```diff
@@ -812,17 +808,16 @@ def _derivative_op(self):
             sage: ModularFormsRing(n=infinity)._derivative_op()
             -X*Y*dX + X*Z*dX + 1/2*Y*Z*dY + 1/4*Z^2*dZ - 1/2*X*dY - 1/4*X*dZ
         """
+        (X, Y, Z, dX, dY, dZ) = self.diff_alg().gens()
```
Comment (kwankyu) [inline src/sage/modular/modform_hecketriangle/abstract_ring.py:812]: No parentheses?

File: src/sage/modular/modform_hecketriangle/abstract_ring.py [hunk 2]
```diff
@@ -812,17 +808,16 @@ def _derivative_op(self):
...
-        else:
-            return 1/self._group.n() * (X*Z-Y) * dX\
-                   + ZZ(1)/ZZ(2) * (Y*Z-X**(self._group.n()-1)) * dY\
-                   + (self._group.n()-2) / (4*self._group.n()) * (Z**2-X**(self._group.n()-2)) * dZ
+                + ZZ(1)/ZZ(2) * (Y*Z-X) * dY\
+                + ZZ(1)/ZZ(4) * (Z**2-X) * dZ
+
+        return 1/self._group.n() * (X*Z-Y) * dX\
+            + ZZ(1)/ZZ(2) * (Y*Z-X**(self._group.n()-1)) * dY\
+            + (self._group.n()-2) / (4*self._group.n()) * (Z**2-X**(self._group.n()-2)) * dZ
```
Comment (kwankyu) [inline src/sage/modular/modform_hecketriangle/abstract_ring.py:820]: How about putting a space before `\` at the end? Here and below.

File: src/sage/modular/modform_hecketriangle/abstract_ring.py [hunk 3]
```diff
@@ -839,17 +834,16 @@ def _serre_derivative_op(self):
             sage: ModularFormsRing(n=infinity)._serre_derivative_op()
             -X*Y*dX - 1/4*Z^2*dZ - 1/2*X*dY - 1/4*X*dZ
         """
+        (X, Y, Z, dX, dY, dZ) = self.diff_alg().gens()
```
Comment (kwankyu) [inline src/sage/modular/modform_hecketriangle/abstract_ring.py:837]: No parentheses?

File: src/sage/modular/modform_hecketriangle/abstract_ring.py (hunk 1 above)
Comment (kwankyu) [reply src/sage/modular/modform_hecketriangle/abstract_ring.py:812]: The same comment here and below.

File: src/sage/modular/modform_hecketriangle/graded_ring.py [hunk 4]
```diff
@@ -57,7 +56,7 @@ def canonical_parameters(group, base_ring, red_hom, n=None):
     return (group, base_ring, red_hom, n)
 
 
-class QuasiMeromorphicModularFormsRing(FormsRing_abstract, CommutativeAlgebra, UniqueRepresentation):
+class QuasiMeromorphicModularFormsRing(FormsRing_abstract, Parent, UniqueRepresentation):
     r"""
```
Comment (kwankyu) [inline src/sage/modular/modform_hecketriangle/graded_ring.py:60]: Do we need `Parent` here? `FormsRing_abstract` is a subclass of `Parent`.

**Comment by fchapoton** [comment]
any other comment ?

**Review by kwankyu — APPROVED** [review]
Perhaps no.

**Comment by fchapoton** [comment]
thanks a lot for your reviews

**Pull request merged**